src/Form/LegacyConsumer/resources/js/conditinal-fields.js

```diff
@@ -0,0 +1,222 @@
+			const inputField = donationForm.querySelector(`[name="${inputFieldName}"]`);
+			const fieldWrapper = inputField.closest('.form-row');
+			const visibilityCondition = visibilityConditions[0]; // Currently we support only one visibility condition.
+			let visible = false;
+			const {field, operator, value} = visibilityCondition;
+
+			const inputs = donationForm.querySelectorAll(`[name="${field}"]`);
+			let hasFieldController = !!inputs.length;
+
+			// Do not apply visibility conditions if field controller does not exit in DOM.
+			if (!hasFieldController) {
+				return;
+			}
+
+			inputs.forEach((input) => {
+				const fieldType = input.getAttribute('type');
+				const comparisonResult = compareWithOperator( operator, input.value, value );
+
+				if (fieldType && (fieldType === 'radio' || fieldType === 'checkbox')) {
+					if (input.checked && comparisonResult ) {
+						visible = true;
+					}
+				} else if ( comparisonResult ) {
+					visible = true;
+				}
+			});
+
+			// Show or Hide field wrapper.
+			visible ?
+				fieldWrapper.classList.remove('give-hidden') :
+				fieldWrapper.classList.add('give-hidden');
+		}
+	}
+
+	/**
+	 * Setup state for condition visibility settings.
+	 * state contains list of watched elements per donation form.
+	 *
+	 * @unreleased
+	 */
+	function addVisibilityConditionsToStateForDonationForm(donationForm) {
+		const uniqueDonationFormId = donationForm.getAttribute('data-id');
+		const watchedFields = getWatchedElementNames(donationForm);
+
+		// Add donation form to state only if visibility conditions exiting for at least form field.
+		if (!uniqueDonationFormId || !Object.entries(watchedFields).length) {
+			return false;
+		}
+
+		state[uniqueDonationFormId] = {
+			watchedElements: watchedFields,
+			...state[uniqueDonationFormId]
+		}
+	}
+
+	/**
+	 * Setup state for condition visibility settings.
+	 * state contains list of watched elements per donation form.
+	 *
+	 * @unreleased
+	 * @returns {Promise<void>}
+	 */
+	function addVisibilityConditionsToStateForAllDonationForm() {
+		document.querySelectorAll('form.give-form')
+			.forEach(function (donationForm) {
+				const uniqueDonationFormId = donationForm.getAttribute('data-id');
+				const watchedFields = getWatchedElementNames(donationForm);
+
+				// Add donation form to state only if visibility conditions exiting for at least form field.
+				if (!uniqueDonationFormId || !Object.entries(watchedFields).length) {
+					return false;
+				}
+
+				state[uniqueDonationFormId] = {
+					watchedElements: watchedFields,
+					...state[uniqueDonationFormId]
+				}
+			});
+	}
+
+	/**
+	 * @unreleased
+	 */
+	function applyVisibilityConditionsAttachedToWatchedField(donationForm, fieldName) {
+		const uniqueDonationFormId = donationForm.getAttribute('data-id');
+
+		// Exit if field is not element of donation form.
+		if (
+			!donationForm ||
+			!uniqueDonationFormId ||
+			!state.hasOwnProperty(uniqueDonationFormId)
+		) {
+			return false;
```

Comment:
Just return. This function doesn't actually return anything, anyway. Same below.

Elaborating, the "return false" idea is popular in WP. I think it's a bad habit. `false` doesn't mean nothing, it means negative, whereas true means positive. So to return false means something. If we simply want to stop at a point, then we should simply return (implicitly void); if we want to return "nothing" then we should use null; if we want to indicate a problem, we should throw an Error/Exception. Returning false should only be used if the function return is _meaningful_ as a boolean, for example `isFormReady()`.

Reply:
@JasonTheAdams Make sense. I refactored code to remove unnecessary `return`: https://github.com/impress-org/givewp/compare/4721eb02efa0...5bcc0c8286a1